homeassistant/components/motionblinds_ble/sensor.py

```diff
@@ -0,0 +1,221 @@
+"""Sensor entities for the Motionblinds BLE integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from math import ceil
+
+from motionblindsble.const import (
+    MotionBlindType,
+    MotionCalibrationType,
+    MotionConnectionType,
+)
+from motionblindsble.device import MotionDevice
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    ATTR_BATTERY,
+    ATTR_CALIBRATION,
+    ATTR_CONNECTION,
+    ATTR_SIGNAL_STRENGTH,
+    CONF_MAC_CODE,
+    DOMAIN,
+    ICON_CALIBRATION,
+    ICON_CONNECTION,
+)
+from .entity import MotionblindsBLEEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+
+@dataclass(frozen=True, kw_only=True)
+class MotionblindsBLESensorEntityDescription(SensorEntityDescription):
+    """Entity description of a sensor entity with native_value attribute."""
+
+    native_value: str | None = None
```

Comment:
This name is misleading, it's actually the initial value. Since you work with assumed state, would `None` not be valid everywhere already as it will render the entities `Unknown`?

Reply:
It would, except for the connect sensor which defaults to `disconnected`

Reply:
Why would a bluetooth device be connected after reboot when the connection is lost?

Reply:
Fair point